apiserversdk/proxy.go

```diff
@@ -26,12 +27,43 @@ func NewMux(config MuxConfig) (*http.ServeMux, error) {
 	if config.Middleware != nil {
 		handler = config.Middleware(proxy)
 	}
+	var eventQueryHandler http.Handler = proxy
+	eventQueryHandler = AddQueryFilterToHandler(eventQueryHandler)
+
 	mux := http.NewServeMux()
 	// TODO: add template features to specify routes.
-	mux.Handle("/apis/ray.io/v1/", handler) // forward KubeRay CR requests.
-	// TODO: add query filters to make sure only KubeRay events are queried.
-	mux.Handle("/api/v1/namespaces/{namespace}/events", handler) // allow querying KubeRay CR events.
+	mux.Handle("/apis/ray.io/v1/", handler)                                // forward KubeRay CR requests.
+	mux.Handle("/api/v1/namespaces/{namespace}/events", eventQueryHandler) // allow querying KubeRay CR events.
 	// TODO: check whether the service belongs to KubeRay first.
 	mux.Handle("/api/v1/namespaces/{namespace}/services/{service}/proxy", handler) // allow accessing KubeRay dashboards and job submissions.
 	return mux, nil
 }
+
+func AddQueryFilterToHandler(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Only allow query requests for events
+		if r.Method != http.MethodGet {
+			http.Error(w, "Only GET requests are allowed for events", http.StatusMethodNotAllowed)
+			return
+		}
```

Comment:
<img width="845" alt="image" src="https://github.com/user-attachments/assets/ef1dd274-fee3-44c7-9f46-6f962740e5eb" />
Just prepend a GET to the route pattern: https://go.dev/blog/routing-enhancements.